deploy/operator/hypershift/deploy_hypershift_cluster.sh

```diff
@@ -121,3 +121,35 @@ echo "Cluster installation has been stopped (either for good or bad reasons)"
 
 KUBECONFIG=$SPOKE_KUBECONFIG wait_for_condition "agentclusterinstall/${ASSISTED_AGENT_CLUSTER_INSTALL_NAME}" "Completed" "90m" "${SPOKE_NAMESPACE}"
 echo "Cluster has been installed successfully!"
+
+
+# Test webhooks
+
+function validate_webhook() {
+  log="$1"
+  pattern="$2"
+  success_msg="$3"
+  failure_msg="$4"
+
+  if [[ $log == *$pattern* ]];
+  then
+    echo $success_msg
+  else
+    echo $failure_msg
+    exit 1
+  fi
+}
+export -f validate_webhook
+
+echo "Test webhooks for InfraEnv successfull update"
+oc --kubeconfig $SPOKE_KUBECONFIG -n $SPOKE_NAMESPACE patch infraenv $ASSISTED_INFRAENV_NAME \
+    -p '{"metadata": {"annotations":{"foo":"bar"}}}' --type merge
+webhook_log=$(oc logs -n $SPOKE_NAMESPACE -l app=agentinstalladmission --since=2s | grep -E 'UPDATE(.)*infraenvs' | tail -1)
+validate_webhook "$webhook_log" "Successful validation" \
+    "Webhook successfull validation for InfraEnv update" "Missing webhook validation"
+
+echo "Test webhooks for InfraEnv failing update"
+error_msg=$(oc --kubeconfig $SPOKE_KUBECONFIG -n $SPOKE_NAMESPACE patch infraenv  $ASSISTED_INFRAENV_NAME \
+    -p '{"spec": {"clusterRef":{"name":"test"}}}' --type merge 2>&1) || true
+validate_webhook "$error_msg" "BadRequest"*"immutable" \
```

Comment:
is it possible putting more of the original message?
(iiuc ``Attempted to change Spec.ClusterRef which is immutable after InfraEnv creation``)

Even if it's a bit long it can give more clarity to the test

Reply:
sure, I'll add more context